### ISSUE TITLE
Incorrect GetHashCode implementation for bindings with float/double

### ISSUE THREAD

**Issue by prasannavl** [mono/SkiaSharp#195]
For the binding definition of `SKSize`, `GetHashCode` is defined as 

```
	public override int GetHashCode ()
		{
			return (int) width ^ (int) height;
		}
```

However, this is incorrect. `Size(10, 10)` and `Size(10.99, 10.99)` will provide the same hash code in this instance.

I have a PR ready, but I presume, this bug also will remain across the definitions. Will probably give it a glance to see if more changes are required with other binding that accept float/double before submitting.

**Comment by mattleibow** [comment]
@prasannavl, thanks for pointing this out.

I had a look at the MS reference source, and they do this:

```csharp
public override int GetHashCode() {
    return base.GetHashCode();
}
```

https://referencesource.microsoft.com/#System.Drawing/commonui/System/Drawing/Advanced/SizeF.cs,232

But in the case of RectangleF:
```csharp
public override int GetHashCode() {
    return unchecked((int)((UInt32)X ^
        (((UInt32)Y << 13) | ((UInt32)Y >> 19)) ^
        (((UInt32)Width << 26) | ((UInt32)Width >>  6)) ^
        (((UInt32)Height <<  7) | ((UInt32)Height >> 25))));
}
```
https://referencesource.microsoft.com/#System.Drawing/commonui/System/Drawing/Advanced/RectangleF.cs,345

Interesting... I am not sure what the best thing is, or even why they differ.

**Comment by mattleibow** [comment]
Mono, on the other had gust casts to an int (I think I copied the mono format):
https://github.com/mono/mono/blob/master/mcs/class/System.Drawing/System.Drawing/SizeF.cs#L263

The rectangle is even more conflicting, they just add all the fields together:
https://github.com/mono/mono/blob/master/mcs/class/System.Drawing/System.Drawing/RectangleF.cs#L502

The question I as is... What should it be?

**Comment by prasannavl** [comment]
Interesting! In retrospect, for some reason, it eluded me that hash code need not be different for different values. It is a 4-byte number after all. It can't be. The only requirement being it has to be the same for the same set of values.

MS's `base.GetHashCode` supposedly XORed the entire memory block 32-bytes at a time, which is technically what mono's implementation does (though I found the results of both to be different - I'm guessing the native implementation does more than just XOR - perhaps changed over time). 

For what it's worth, I think the mono implementation is simpler, and uses lesser instructions (though negligible) than MS, in case of the RectangleF.

This is an interesting tid-bit, though I think now that this whole issue isn't actually one. I apologize, if I had wasted your time unintentionally. Closing this.
